### PR TITLE
NAS-127070 / 24.10 / `core.bulk` and roles

### PR DESCRIPTION
**Pull request by themylogin**
* Allow unauthorized access to core.bulk
* Authenticate calls in `core.bulk` and pass `app`
* Audit calls made with `core.bulk`

### CODE REVIEW

**Comment by themylogin** [comment]
https://ci.tn.ixsystems.net/jenkins/job/TrueNAS%20SCALE%20-%20Unstable/job/API%20Tests%20-%20TrueNAS%20SCALE%20with%20middleware%20reinstalled%20from%20a%20branch%20Incremental/1657/

**Review by yocalebo — APPROVED** [review]
The link to the CI that actually ran is here: [here](https://ci.tn.ixsystems.net/jenkins/job/TrueNAS%20SCALE%20-%20Unstable/job/API%20Tests%20-%20TrueNAS%20SCALE%20with%20middleware%20reinstalled%20from%20a%20branch%20Incremental/1659/#showFailuresLink)

**Comment by yocalebo** [comment]
Please wait for @anodos325 review before merge.

**Pull request merged**
